src/microbe_stage/player_hover_info.cpp

```diff
@@ -137,33 +137,29 @@ void
             }
 
             // TODO: this would be nice to verify just once during startup
-            if(microbeComponent->GetPropertyTypeId(0) != stringType ||
-                std::strncmp(microbeComponent->GetPropertyName(0),
-                    "speciesName", sizeof("speciesName") - 1) != 0) {
+            if(microbeComponent->GetPropertyTypeId(0) != speciesType) {
 
                 LOG_ERROR("PlayerHoverInfoSystem: Run: MicrobeComponent object "
                           "doesn't "
-                          "have \"string speciesName\" as the first property");
+                          "have \"Species@ species\" as the first property");
                 continue;
             }
 
-            const auto* name = static_cast<std::string*>(
+            const auto** speciesHandle = static_cast<const Species**>(
                 microbeComponent->GetAddressOfProperty(0));
 
-            const auto map = world.GetPatchManager().getCurrentMap();
-            const auto species =
-                map ? map->findSpeciesByName(*name) : Species::pointer{};
+            const Species* species = *speciesHandle;
 
-
-            if(species) {
+            if(species != nullptr) {
```

Comment:
I know I told you to do it like this, but keeping the old check is better (explicit comparison to nullptr should be avoided).